haystack/utils/openai_utils.py

```diff
@@ -22,38 +22,20 @@
 machine = platform.machine().lower()
 system = platform.system()
 
-
 OPENAI_TIMEOUT = float(os.environ.get(HAYSTACK_REMOTE_API_TIMEOUT_SEC, 30))
 OPENAI_BACKOFF = int(os.environ.get(HAYSTACK_REMOTE_API_BACKOFF_SEC, 10))
 OPENAI_MAX_RETRIES = int(os.environ.get(HAYSTACK_REMOTE_API_MAX_RETRIES, 5))
 
 
-USE_TIKTOKEN = False
-if sys.version_info >= (3, 8) and (machine in ["amd64", "x86_64"] or (machine == "arm64" and system == "Darwin")):
-    USE_TIKTOKEN = True
-
-if USE_TIKTOKEN:
-    import tiktoken  # pylint: disable=import-error
-    from tiktoken.model import MODEL_TO_ENCODING, MODEL_PREFIX_TO_ENCODING
-else:
-    logger.warning(
-        "OpenAI tiktoken module is not available for Python < 3.8,Linux ARM64 and AARCH64. Falling back to GPT2TokenizerFast."
-    )
-
-
 def load_openai_tokenizer(tokenizer_name: str):
     """Load either the tokenizer from tiktoken (if the library is available) or fallback to the GPT2TokenizerFast
     from the transformers library.
 
     :param tokenizer_name: The name of the tokenizer to load.
     """
-    if USE_TIKTOKEN:
-        logger.debug("Using tiktoken %s tokenizer", tokenizer_name)
-        tokenizer = tiktoken.get_encoding(tokenizer_name)
-    else:
-        logger.debug("Using GPT2TokenizerFast tokenizer")
-        tokenizer = GPT2TokenizerFast.from_pretrained(tokenizer_name)
-    return tokenizer
+
+    logger.debug("Using tiktoken %s tokenizer", tokenizer_name)
+    return tiktoken.get_encoding(tokenizer_name)
 
 
 def count_openai_tokens(text: str, tokenizer) -> int:
```

Comment:
We can probably remove this function altogether

Reply:
Alright, that's done now. 🙂